sdk/cargo-build-bpf/src/main.rs

```diff
@@ -142,26 +250,89 @@ fn build_bpf_package(config: &Config, target_directory: &Path, package: &cargo_m
         println!("Legacy program feature detected");
     }
 
-    let xargo_build = config.bpf_sdk.join("rust").join("xargo-build.sh");
-    let mut xargo_build_args = vec![];
+    let criterion_version = if cfg!(target_os = "macos") {
+        "v2.3.2"
+    } else {
+        "v2.3.3"
+    };
+    let criterion_filename = if cfg!(target_os = "macos") {
+        "criterion-v2.3.2-osx-x86_64.tar.bz2"
+    } else {
+        "criterion-v2.3.3-linux-x86_64.tar.bz2"
+    };
+    install_if_missing(
+        &config,
+        "criterion",
+        criterion_version,
+        "https://github.com/Snaipe/Criterion/releases/download",
+        &PathBuf::from(criterion_filename),
+    )
+    .expect("Failed to install criterion");
+    let bpf_tools_filename = if cfg!(target_os = "macos") {
```

Comment:
For windows we can just error at this point

Reply:
Can a BPF toolchain exist on Windows in principle (from some other source, not Solana)?  If it exists and we only need to link to it, maybe it's better to error at the point where it's determined that we need to download the bpf-tools package for Windows, which we can't.

Reply:
It probably can't exist yet.  We build `cargo-build-bpf` for windows right now just to help Windows users discover that they need a Linux or macOS machine faster.  Otherwise `cargo build-bpf` fails in more mysterious ways.

So `cargo-build-bpf` successfully building for windows but failing at runtime is just fine for now

Reply:
added an exit on windows at the beginning of main()